src/lib.rs

```diff
@@ -421,24 +443,30 @@ mod tests {
 
     #[test]
     fn test_mime_from_str() {
-        assert_eq!(FromStr::from_str("text/plain"), Ok(mime!(Text/Plain)));
-        assert_eq!(FromStr::from_str("TEXT/PLAIN"), Ok(mime!(Text/Plain)));
-        assert_eq!(FromStr::from_str("text/plain; charset=utf-8"), Ok(mime!(Text/Plain; Charset=Utf8)));
-        assert_eq!(FromStr::from_str("text/plain;charset=\"utf-8\""), Ok(mime!(Text/Plain; Charset=Utf8)));
-        assert_eq!(FromStr::from_str("text/plain; charset=utf-8; foo=bar"),
-            Ok(mime!(Text/Plain; Charset=Utf8, ("foo")=("bar"))));
+        assert_eq!(Mime::from_str("text/plain").unwrap(), mime!(Text/Plain));
```

Comment:
Technically, these changes aren't needed. They were until I add `Vec` as the default for `T`, but I think it reads better anyway.